hebiros_gazebo_plugin/sim/joint.h

```diff
@@ -0,0 +1,138 @@
+#pragma once
+
+#include <memory>
+
+#include "Eigen/Dense"
+
+#include "pid_controller.h"
+#include "temperature_model.h"
+#include "temperature_safety_controller.h"
+
+namespace hebi {
+namespace sim {
+
+// Generally, each "OnUpdate" call of the main loop, the joint should:
+// (1) have any commands set w/ `setCommand` (this can be called multiple times per cycle; only
+// one will end up taking effect, depending on which senders are locked out)
+// (2) have the `update` function run, which generates its PWM command
+// (3) have its feedback set
+class Joint : public std::enable_shared_from_this<Joint> {
+
+public:
+  enum class JointType {
+    X5_1, X5_4, X5_9,
+    X8_3, X8_9, X8_16
+  };
+
+  enum class ControlStrategy {
+    Off = 0,
+    DirectPWM = 1,
+    Strategy2 = 2,
+    Strategy3 = 3,
+    Strategy4 = 4
+  };
+
+  using SimTime = double;
+
+  static std::unique_ptr<Joint> tryCreate(const std::string& family, const std::string& name, const std::string& type);
+
+  void updateImu(const Eigen::Vector3f& accelerometer, const Eigen::Vector3f& gyro);
+  const Eigen::Vector3f getAccelerometer() { return accelerometer_; }
```

Comment:
Did you mean to make these const refs? Not a big deal, but that is what I'm assuming you meant.

Reply:
good catch -- yeah, the const is absolutely pointless here without the ref.